module/strategy/WalkToFieldPosition/src/WalkToFieldPosition.cpp

```diff
@@ -40,21 +41,47 @@ namespace module::strategy {
     using message::input::Sensors;
     using message::localisation::Field;
     using message::planning::WalkTo;
+    using message::skill::Walk;
     using WalkToFieldPositionTask = message::strategy::WalkToFieldPosition;
 
     WalkToFieldPosition::WalkToFieldPosition(std::unique_ptr<NUClear::Environment> environment)
         : BehaviourReactor(std::move(environment)) {
 
         on<Configuration>("WalkToFieldPosition.yaml").then([this](const Configuration& config) {
             // Use configuration here from file WalkToFieldPosition.yaml
-            this->log_level = config["log_level"].as<NUClear::LogLevel>();
+            this->log_level       = config["log_level"].as<NUClear::LogLevel>();
+            cfg.stop_threshold    = config["stop_threshold"].as<double>();
+            cfg.stopped_threshold = config["stopped_threshold"].as<double>();
         });
 
+        on<Start<WalkToFieldPositionTask>>().then([this] { current_threshold = cfg.stop_threshold; });
+
         on<Provide<WalkToFieldPositionTask>, With<Field>, With<Sensors>>().then(
             [this](const WalkToFieldPositionTask& walk_to_field_position, const Field& field, const Sensors& sensors) {
                 // Transform from desired field position into robot space
-                Eigen::Isometry3d Hrd = sensors.Hrw * field.Hfw.inverse() * walk_to_field_position.Hfd;
-                emit<Task>(std::make_unique<WalkTo>(Hrd));
+                Eigen::Isometry3d Hrd      = sensors.Hrw * field.Hfw.inverse() * walk_to_field_position.Hfd;
+                double translational_error = Hrd.translation().norm();
+
+                // Compute the error between the desired heading and the measured heading
+                Eigen::Isometry3d Hfr          = field.Hfw * sensors.Hrw.inverse();
+                Eigen::Vector3d desired_unit_x = walk_to_field_position.Hfd.linear().col(0);
+                double desired_heading         = std::atan2(desired_unit_x.y(), desired_unit_x.x());
+                double measured_heading        = std::atan2(Hfr.linear().col(0).y(), Hfr.linear().col(0).x());
+
+                double angle_error = std::abs(desired_heading - measured_heading);
+                // Normalize the angle error to be within the range [-pi, pi]
+                angle_error = std::atan2(std::sin(angle_error), std::cos(angle_error));
+
+                if (translational_error < current_threshold && std::abs(angle_error) < current_threshold
+                    && !walk_to_field_position.dont_stop_at_target) {
+                    emit<Task>(std::make_unique<Walk>(Eigen::Vector3d::Zero()));
+                    current_threshold = cfg.stopped_threshold;
```

Comment:
probably add some comments around this stop/stopping concept, or maybe just put it in the readme